### ISSUE TITLE
Better Updating of Renter Persistence

### ISSUE THREAD

**Issue by eddiewang** [NebulousLabs/Sia#2256]
Want to put this idea down here while it's fresh.
I recently migrated to a new Macbook Pro (yay), and copied my renter folder over.
Since I was playing around with setting up siad again, I vim'd into the renter folder to read some of the logs.

There are some easy fixes that can be implemented to the Renter Persistence file.
![image](https://user-images.githubusercontent.com/2057997/29664914-ade34f40-889f-11e7-9004-0760fa441173.png)

1. Renter tracking files that have been deleted
Maybe there is an underlying reason to do this, but the renter persistence file seems to continue to track files that have been long deleted either through Sia-UI or siac. We should clean these files up as they are no longer necessary.

2. Update repair path as file downloads
The repair path seems to be set upon first upload, and it is no longer changed. I'm not sure how the online repair works, but I am assuming the file is downloaded to cache, repaired by uploading to new contracts, and then deleted from cache. We should be updating the location of the file everytime the file is downloaded.

In my case, I've redownloaded all the files I've stored on Sia into a new folder on my computer, and deleted the old files. The renter log now shows that it's having problems repairing the files through the renter:
![image](https://user-images.githubusercontent.com/2057997/29665037-2a70c8ee-88a0-11e7-9ccf-946e884737b0.png)

I'm not sure if these glitches are happening because of the version updates, but we should have a cleanup next update that properly syncs the persistence file so it can better handle file repairs.

**Comment by tbenz9** [comment]
Seems like a bug to me.

Recommendation to devs: tag as 'bug'

**Comment by lukechampine** [comment]
Agreed on point 1. As for point 2, I think we'd want something akin to what BitTorrent clients do: let the user specify the new location of the file (or a folder of files).

**Comment by MSevey** [comment]
Point 1 will be addressed with the new file format #2204
